src/gen_rules.ml

```diff
@@ -359,6 +359,23 @@ module type Gen = sig
   val sctx : Super_context.t
 end
 
+let stanza_package = function
+  | Library { public = Some { package; _ }; _ }
+  | Alias { package = Some package ;  _ }
+  | Install { package; _ }
+  | Documentation { package; _ }
+  | Tests { package = Some package; _} ->
+      Some package
+  | _ -> None
+
+let relevant_stanzas pkgs stanzas =
+  let is_relevant stanza =
+    match stanza_package stanza with
+    | Some package -> Package.Name.Set.mem pkgs package.name
+    | None -> true
+  in
+  List.filter stanzas ~f:is_relevant
```

Comment:
Personally, the intermediate `is_relevant` name doesn't really increase readability here.

Reply:
right, it was on toplevel for a moment. I'll inline it back.